test/check-machines-filesystems

```diff
@@ -60,8 +60,12 @@ class TestMachinesFilesystems(machineslib.VirtualMachinesCase):
 
         b.set_input_text("#vm-subVmTest1-filesystems-modal-mountTag", "")
         b.set_input_text("#vm-subVmTest1-filesystems-modal-source-group input", "/tmp")
-        b.wait_visible("#vm-subVmTest1-filesystems-modal-source")
+        # In some hosts with high latency, the path dropdown list needs some time to load, the check will fail if it is too quick.
+        # Thus, add more check to ensure that the loading is finished
+        b.wait_in_text("#vm-subVmTest1-filesystems-modal-source", "/tmp")
         b.click("#vm-subVmTest1-filesystems-modal-source button:nth-child(1)")
+        b.wait_not_present("#vm-subVmTest1-filesystems-modal-source")
+        b.wait_attr_contains("#vm-subVmTest1-filesystems-modal-source-group input", "value", "/tmp")
```

Comment:
Can you please explain this? The "click" could really do with a comment what it does, is that the help icon button for "More info for source path field"? But why would clicking it make the whole "Source path" input go away? And then *after* you wait for it to go away you check one of its attributes?

(Note: it may well be right: It is just absolutely impossible for me to understand it without being intimately familiar with what this does)

Reply:
So I just want to make sure the dropdown list disappears(b.wait_not_present) and the input accepts the text correctly(b.wait_attr_contains). Then, after some new trying, I think b.wait_attr_contains is not necessary as set_input_text has already done this, but according to the previous comment I left, I think b.wait_not_present is helpful to ensure that the dropdown list has disappeared. Do you think that makes sense?

Reply:
Right, the `wait_attr_contains` is the same as `.val()`, and set_input_text already does the .wait_val() for this. The wait_not_present() strategically makes sense, I just have *no* idea what `-modal-source` actually is -- is that really specifically the *dropdown menu* part of the file selector? The code says

    <FileAutoComplete id={`${idPrefix}-modal-source`}

which should be the text `<input>` and sometimes the menu (which can open or close), and this is specifically meant for the *menu* part.

That's what I mean with "please explain this" -- and I don't mean "here in github", but "please add a comment".  These selectors are not understandable (which isn't your fault, but it still needs comments to make sense).

Thanks!

Please also rebase this, the PR is quite old by now.

Reply:
OK, I will rebase and add some comments